File: Java/Loon-Neo/src/loon/utils/ArrayByteOutput.java
package loon.utils;

import java.io.IOException;
import java.io.OutputStream;

import loon.LRelease;

public class ArrayByteOutput extends OutputStream implements LRelease {

	private final ArrayByte _buffer;

	public ArrayByteOutput(int size) {
		_buffer = new ArrayByte(size);
	}

	public ArrayByteOutput() {
		this(8192 * 10);
	}

	public ArrayByte getArrayByte() {
		return _buffer;
	}

	public byte[] toByteArray() {
		return _buffer.getBytes();
	}

	@Override
	public void write(int b) throws IOException {
		_buffer.writeByte(b);
	}

	@Override
	public void close() {
		_buffer.close();
	}

}

File: C#/Loon2MonoGame/LoonMonoGame-Lib/loon/utils/ArrayByteOutput.cs
using java.io;

namespace loon.utils
{
    public class ArrayByteOutput : OutputStream, LRelease
    {

        private readonly ArrayByte _buffer;

        public ArrayByteOutput(int size)
        {
            _buffer = new ArrayByte(size);
        }

        public ArrayByteOutput() : this(8192 * 10)
        {

        }

        public ArrayByte GetArrayByte()
        {
            return _buffer;
        }

        public sbyte[] ToByteArray()
        {
            return _buffer.GetBytes();
        }

        public override void Write(int b)
        {
            _buffer.WriteByte(b);
        }

        public override void Close()
        {
            _buffer.Close();
        }

    }

}
